sympy/sets/tests/test_sets.py

```diff
@@ -1052,3 +1052,9 @@ def test_issue_11174():
 
     soln = Intersection(S.Reals, FiniteSet(x), evaluate=False)
     assert Intersection(FiniteSet(x), S.Reals) == soln
+
+def test_issue_9577():
+    un = S.UniversalSet
+    assert Union(Complement(un, Interval(1, 2)), Interval(3, 4)) == \
+            Complement(un, Interval(1, 2))
+    assert Union(Complement(un, Interval(1, 2)), Interval(1, 3)) == un
```

Comment:
@moorepants : @madan96 is the author of this PR. I think it is better to ping him here.